compiler/expressions.go

```diff
@@ -982,6 +982,9 @@ func (fc *funcContext) translateBuiltin(name string, sig *types.Signature, args
 	case "len":
 		switch argType := fc.typeOf(args[0]).Underlying().(type) {
 		case *types.Basic:
+			if containsBinaryExpr(args[0]) {
```

Comment:
Oh, that's a good point. I had through about all top level combinations I could think of but totally didn't think of a binary expression used as an argument or indexer (e.g. `len(str[0 : i+2])`). Hmmm, the extra parenthesis wouldn't be too bad but it would be better to only have it hit only if it is at the root of the expression.

I'm going to experiment some but once you said that, I think we could simply do `if _, ok := args[0].(*ast.BinaryExpr); ok {` instead of `containsBinaryExpr`. I think in my attempt to make it more robust, I missed the obvious.

Reply:
Yep, over complicated the fix. It works for all the cases I could think of, including those you pointed out, using simply the type check. I pushed up the fix